crates/test/macros/src/lib.rs

```diff
@@ -140,6 +140,17 @@ fn tests_runner_impl(args: TokenStream) -> parse::Result<TokenStream> {
                 }
             }
         }
+
+        #[cfg(all(test, miri))]
+        #[start]
+        fn main(argc: isize, argv: *const *const u8) -> isize {
+            let ret = __test_runner::main(::core::ptr::null_mut());
+
+            // Clean up app state.
+            ::flipperzero_rt::__macro_support::__wait_for_thread_completion();
+
+            ret.try_into().unwrap()
```

Comment:
Since the only case for failing here is `size_of::<isize>` being `<= size_of::<i32>`, maybe there is a more explicit way to handle this, such as casting (we actually only care about the code being either `0` ot non-`0`, I guess)